### PR TITLE
notebook magic used old terminology "input"

### LINKED ISSUE #8028: cannot export to notebook with magic command in IPython 3.0.0
I was trying to export command history into a notebook with `%notebook -e $notebook_file`, and got the following error:
<pre><code>/Users/w/python-venv/2.7/lib/python2.7/site-packages/IPython/core/magics/basic.pyc in notebook(self, s)
    605                 cells.append(v4.new_code_cell(
    606                     execution_count=execution_count,
--> 607                     source=source
    608                 ))
    609             nb = v4.new_notebook(cells=cells)

NameError: global name 'source' is not defined</pre></code>

Had a quick look in the file `IPython/core/magics/basic.py`, and I guess the line `source=source` should be replaced by `source=input`?

Here is the output of `python -c "import IPython; print(IPython.sys_info())"` on my system:
<pre><code>{'commit_hash': u'f75fda4',
 'commit_source': 'installation',
 'default_encoding': 'UTF-8',
 'ipython_path': '/Users/w/python-venv/2.7/lib/python2.7/site-packages/IPython',
 'ipython_version': '3.0.0',
 'os_name': 'posix',
 'platform': 'Darwin-14.1.0-x86_64-i386-64bit',
 'sys_executable': '/Users/w/python-venv/2.7/bin/python',
 'sys_platform': 'darwin',
 'sys_version': '2.7.6 (default, Sep  9 2014, 15:04:36) \n[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)]'}</pre></code>

### PR DESCRIPTION
**Pull request by Carreau**
replace by 'source'.

Fixes #8028

### CODE REVIEW

**Comment by takluyver** [comment]
Tests plz ;-)

**Comment by Carreau** [comment]
Yeah...

**Comment by Carreau** [comment]
First test (14c61a) should fail.

File: IPython/core/tests/test_magic.py [hunk 1]
```diff
@@ -623,9 +623,13 @@ def test_extension():
 @skipIf(dec.module_not_available('IPython.nbformat'), 'nbformat not importable')
 class NotebookExportMagicTests(TestCase):
     def test_notebook_export_json(self):
+        _ip = get_ipython()
+        _ip.history_manager.reset()   # Clear any existing history.
+        cmds = [u"a=1", u"def b():\n  return a**2", u"print(a, b())"]
```
Comment (takluyver) [inline IPython/core/tests/test_magic.py]: Can you stick a non-ascii character in one of these as well, to check that it's not broken with unicode.

Comment (Carreau) [inline IPython/core/tests/test_magic.py]: Fixed. I miss rebased so the test  of previous commit should pass, too lazy to split in commit in several ones;

**Pull request merged**